### ISSUE TITLE
Ability to extend Array in circular reference

### ISSUE THREAD

**Issue by JorritPosthuma** [vega/ts-json-schema-generator#83]
Hi! I 'm having some trouble with generating JSON schema for types with circular references. I would like to generate the schema for the following:

```typescript
export type Tag = TagPrimitive | Tag[];
export type TagPrimitive = string | number | boolean | null;
```

However, as this isn't valid TypeScript, it is common to change it to the following:

```typescript
export type Tag = TagPrimitive | TagArray;
export type TagPrimitive = string | number | boolean | null;
export interface TagArray extends Array<Tag> {}
```

This however results in the following error: `Cannot read property 'getId' of undefined`. So, the first version does generate the proper JSON schema, while the second is valid typescript. Is it possible to support the second method as well?

**Comment by domoritz** [comment]
I'd be happy to review a pull request. I don't see any fundamental limitation.

**Comment by kayahr** [comment]
The `Cannot read property 'getId' of undefined` error was fixed by #107 so the generator should be able to output a JSON schema since version 0.45. 

But unfortunately the output is not correct:

```javascript
{
  "$ref": "#/definitions/TagArray",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "TagArray": {
      "additionalProperties": {
        "$ref": "#/definitions/Tag"
      },
      "properties": {
        "length": {
          "description": "Gets or sets the length of the array. This is a number one higher than the highest element defined in an array.",
          "type": "number"
        }
      },
      "required": [
        "length"
      ],
      "type": "object"
    },
    "TagPrimitive": {
      "type": [
        "string",
        "number",
        "boolean",
        "null"
      ]
    }
  }
}
```

So `interface TagArray extends Array<Tag>` is not parsed into a normal array. Instead it is parsed into an Object extending the Array class requiring the `length` property. And the definition `#/definitions/Tag` is not even written to the schema.
